src/backend/commands/analyze.c

```diff
@@ -933,13 +914,12 @@ static void analyzeRelation(Relation relation, List *lAttributeNames, bool rooto
 	else
 	{
 		initStringInfo(&err_msg);
-		gp_statistics_estimate_reltuples_relpages_external_pxf(relation, &location, &estimatedRelTuples, &estimatedRelPages, &err_msg);
+		analyzePxfEstimateReltuplesRelpages(relation, &location, &estimatedRelTuples, &estimatedRelPages, &err_msg);
 		if (err_msg.len > 0)
 		{
-			ereport(WARNING,
-					(errmsg("skipping \"%s\" --- error returned: %s",
-							RelationGetRelationName(relation),
-							err_msg.data)));
+			elog(ERROR,
+					"%s",
+					err_msg.data);
```

Comment:
No, only analyze for this table is aborted - each analyzeRelation is encapsulated inside a subtransaction.